presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcFileRewriter.java

```diff
@@ -413,9 +428,225 @@ public void testUncompressedSize(boolean useOptimizedOrcWriter)
         }
 
         File newFile = new File(temporary, randomUUID().toString());
-        OrcFileInfo info = new OrcRecordFileRewriter().rewrite(file, newFile, new BitSet());
+        OrcFileInfo info = createFileRewriter(useOptimizedOrcWriter).rewrite(getColumnTypes(columnIds, columnTypes), file, newFile, new BitSet());
         assertEquals(info.getRowCount(), 3);
-        assertEquals(info.getUncompressedSize(), 55);
+        assertBetweenInclusive(info.getUncompressedSize(), 55L, 55L * 2);
+    }
+
+    /**
+     * The following test add or drop different columns
+     */
+    @Test(dataProvider = "useOptimizedOrcWriter")
```

Comment:
I think we don't need to test the old orc writer here as it's not aware of any columns.

Reply:
As discussed offline, let's keep it to raise the awareness of the buggy legacy rewriter.